PokeApiNet/PokeApiClient.cs

```diff
@@ -74,6 +74,23 @@ public PokeApiClient(HttpMessageHandler messageHandler, ProductHeaderValue userA
             _client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(userAgent));
         }
 
+        /// <summary>
+        /// Construct accepting directly a HttpClient. Useful when used in projects where IHttpClientFactory is used to create and configure HttpClient instances with different policies.
+        /// See https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
+        /// </summary>
+        /// <param name="httpClient"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public PokeApiClient(HttpClient httpClient)
```

Comment:
I was going to mention that we need to set the user agent string to match what the other constructors are doing, but [it looks like that requirement was removed in a recent commit to the PokeAPI docs page](https://github.com/PokeAPI/pokeapi.co/pull/80/files#diff-94b6c17a6dfe07cbce8c054ca835ebbc662788b9dd706142f805f552887e87a0L159).